src/include/migraphx/matcher.hpp

```diff
@@ -582,6 +584,52 @@ inline auto outputs()
     };
 }
 
+inline auto trace(std::string s)
```

Comment:
what's the use case for this one ?

Reply:
1) To see whats get passed to the matcher(ie to see how far it makes it in the matcher).
2) To see the context of what is matching. You can place it on the outer matcher while you have traces on the inner matcher: `trace("outer")(some_matcher(trace("inner")(another_matcher)))`.